packages/mds-transactions-service/tests/service.spec.ts

```diff
@@ -22,28 +22,81 @@ describe('Transaction Repository Tests', () => {
 
 const TransactionServer = TransactionServiceManager.controller()
 
+const device_id = 'ee6bf5c7-bce0-46c9-a5c9-8652724059d7'
+const provider_id = '3452fa87-bfd7-42c5-9c53-5e07bde13671'
+const transaction_id = '37bd96ac-69bd-4634-9b22-ff081d7a5a09'
+const receipt_id = 'a5eb612e-a154-4339-a760-aee95908dc51'
+
+const receipt = { receipt_id, timestamp: Date.now(), receipt_details: {}, origin_url: '' }
+
+const malformed_uuid = '176b8453-ccaf-41c7-a4df-f7b3f80bddd1xxxxxxx'
+
 describe('Transaction Service Tests', () => {
   beforeAll(async () => {
     await TransactionServer.start()
   })
 
-  it('Post Transaction', async () => {
-    const transaction = await TransactionServiceClient.createTransaction({ name: 'Test Transaction', text: 'This is a test' })
-    expect(transaction.name).toEqual('Test Transaction')
+  it('Post Good Transaction', async () => {
+    const transaction = await TransactionServiceClient.createTransaction({
+      transaction_id,
+      provider_id,
+      device_id,
+      timestamp: Date.now(),
+      amount: 100, // "I'd buy THAT for a dollar!"
+      fee_type: 'base_fee',
+      receipt
+    })
+    expect(transaction.device_id).toEqual(device_id)
+    expect(transaction.transaction_id).toEqual(transaction_id)
+  })
+
+  it('Post Transaction with malformed transaction_id', async () => {
+    try {
+      await TransactionServiceClient.createTransaction({
+        transaction_id: malformed_uuid,
+        provider_id,
+        device_id,
+        timestamp: Date.now(),
+        amount: 100, // "I'd buy THAT for a dollar!"
+        fee_type: 'base_fee',
+        receipt
+      })
+      expect('did not happen').toBe('happened')
```

Comment:
🤔

Reply:
From https://jestjs.io/docs/en/tutorial-async:

```
// Or using async/await with `.rejects`.
it('tests error with async/await and rejects', async () => {
  expect.assertions(1);
  await expect(user.getUserName(3)).rejects.toEqual({
    error: 'User with 3 not found.',
  });
});
```

Reply:
+1, Jest makes it pretty painless to test promise rejections.

Reply:
eh, this doesn't read better to me

Reply:
No point in debating about readability, but _functionally_ the solution that Jest offers is vastly superior. Being able to test that a particular call rejects, and test that it rejects with a _particular error_ is far better than just adding a falsy assertion that should be unreachable. 

Also, the falsy assertion approach would not work in a scenario where you have one test that is testing multiple rejections, e.g.
```
it('some test', async () => {
  await expectToThrow()
  await expectThisThrowsToo()
  expect(0).toEqual(1)
}
```
You'd end up aborting at the first throw, and not testing the second at all.